src/main/scala/viper/gobra/frontend/info/implementation/typing/StmtTyping.scala

```diff
@@ -137,6 +137,22 @@ trait StmtTyping extends BaseTyping { this: TypeInfoImpl =>
     case _: PSeq => noMessages
     case _: PEmptyStmt => noMessages
 
+    case n: POutline =>
+      error(n.spec, "Currently, outline statements do not support the pure annotation.", n.spec.isPure) ++ 
+      error(n.spec, "Currently, the old keyword is not supported", (n.spec.posts ++ n.spec.pres).find{exp => 
+        allChildren(exp).find{case _: POld => true case _ => false}.isDefined
+      }.isDefined) ++
+      n.body.flatMap{stmt => 
+        val children = allChildren(stmt)
+        children.collect{
+          case _: POld => error(stmt, "Currently, the old keyword is not supported")
+          case _: PReference => error(stmt, "Currently, taking the reference is not supported")
+        }.flatten
+      } ++
+      n.body.collect{case stmt: PReturn => 
+          error(stmt, "Using return in an outlined block is not allowed")
+      }.flatten
```

Comment:
I think that this does not catch return statements inside of compound statements, e.g. while.